provider/rancher.go

```diff
@@ -320,6 +323,27 @@ func listRancherServices(client *rancher.RancherClient) []*rancher.Service {
 	return servicesList
 }
 
+func listRancherPublicEndponts(client *rancher.RancherClient) map[string]string {
+
+	var publicEndpointMap = make(map[string]string)
+
+	hosts, err := client.Host.List(nil)
+
+	if err != nil {
+		log.Errorf("Cannot get Rancher Endpoints %+v", err)
```

Comment:
Please [decapitalize the first word of the error message](https://github.com/golang/go/wiki/CodeReviewComments#error-strings).